### PR TITLE
fix: fallback to ssh if https fails and vice versa

### LINKED ISSUE #2323: Fallback to ssh if git https doesn't work
Currently you can only define a git dependency either via https:// or git@ (ssh). Would be great if DevSpace would fallback to try https in the ssh case and ssh in the https case.

So the logic would be for `https`:

* Try [https://domain.com/path](https://domain.com/path), if that doesn't work, use git@domain:path

For ssh the logic would be the reverse

### PR DESCRIPTION
**Pull request by pratikjagrut**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #2323 

**Please provide a short message that should be published in the DevSpace release notes**
Fixed an issue where DevSpace hangs till git credentials are not provided.

### CODE REVIEW

File: pkg/util/git/git.go [hunk 1]
```diff
@@ -64,7 +65,14 @@ func (gr *GitCLIRepository) Clone(ctx context.Context, options CloneOptions) err
 		}
 
 		args = append(args, options.Args...)
-		out, err := command.CombinedOutput(ctx, gr.LocalPath, expand.ListEnviron(os.Environ()...), "git", args...)
+		os.Setenv("GIT_SSH_COMMAND", "ssh -oBatchMode=yes")
```
Comment (lizardruss) [inline pkg/util/git/git.go:68]: Is this to prevent prompting for a password? If not, a comment with explanation might be helpful.

**Review by lizardruss — COMMENTED** [review]
Nice work!

I had some questions that might prompt changes, but otherwise it looks good.

File: pkg/util/git/git.go [hunk 2]
```diff
@@ -64,7 +65,14 @@ func (gr *GitCLIRepository) Clone(ctx context.Context, options CloneOptions) err
...
+		gitEnv := []string{
+			"GIT_TERMINAL_PROMPT=0",
+			"GIT_SSH_COMMAND='ssh -oBatchMode=yes'",
+		}
+		gitEnv = append(gitEnv, os.Environ()...)
+		out, err := command.CombinedOutput(ctx, gr.LocalPath, expand.ListEnviron(gitEnv...), "git", args...)
+		os.Unsetenv("GIT_SSH_COMMAND")
```
Comment (lizardruss) [inline pkg/util/git/git.go:75]: Is `os.Setenv()` and `os.Unsetenv()` required, or is passing `gitEnv` when running the command sufficient?

File: pkg/util/git/git.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/util/git/git.go:68]: Yes, this is for preventing prompts and preventing ssh from asking for adding unknown hosts to known host list. I'll add comments there.

File: pkg/util/git/git.go (hunk 2 above)
Comment (pratikjagrut) [reply pkg/util/git/git.go:75]: It's required. `GIT_SSH_COMMAND ` will be used by the underlying ssh command.

File: pkg/util/git/git.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/util/git/git.go:70]: Done.

File: pkg/util/git/git.go (hunk 2 above)
Comment (pratikjagrut) [reply pkg/util/git/git.go]: Yes, it is, I seem to misunderstand it.

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**